### ISSUE TITLE
Error message when loading .RProj from remote server 

### ISSUE THREAD

**Issue by baderstine** [rstudio/rstudio#5000]
New error just started occurring with latest version of RStudio on Mac:

### System details

    RStudio Edition : Desktop 
    RStudio Version : Version 1.2.1335 Build 1351 (f1ac3452)
    R version 3.5.3 (2019-03-11)
    Platform: x86_64-apple-darwin15.6.0 (64-bit)
    Running under: macOS Mojave 10.14.5

### Steps to reproduce the problem  

1. Map a remote server.  Finder -> Go -> Connect to Server...   
2. Open a .RProj file from remote server   

### Describe the problem in detail
RStudio displays error: 

> This site can’t be reached
> 127.0.0.1 refused to connect.
> Try:
> 
> Checking the connection
> Checking the proxy and the firewall
> ERR_CONNECTION_REFUSED
> Reload  (button)
> DETAILS (link)

Click RELOAD and the project loads normally.  

### Describe the behavior you expected
Load the remote project file without an error message.  

Three error messages are written to the log when this behavior occurs:  
```
24 Jun 2019 16:24:51 [rsession-<user>] ERROR system error 45 (Operation not supported) [lock-file=<remote-path>/.Rproj.user/442D8A65/sources/s-782BF86F/lock_file]; OCCURRED AT: virtual bool rstudio::core::AdvisoryFileLock::isLocked(const rstudio::core::FilePath &) const /Users/vagrant/workspace/IDE/macos-v1.2/src/cpp/core/file_lock/AdvisoryFileLock.cpp:85; LOGGED FROM: virtual bool rstudio::core::AdvisoryFileLock::isLocked(const rstudio::core::FilePath &) const /Users/vagrant/workspace/IDE/macos-v1.2/src/cpp/core/file_lock/AdvisoryFileLock.cpp:87
24 Jun 2019 16:24:51 [rsession-<user>] ERROR system error 45 (Operation not supported) [lock-file=<remote-path>/.Rproj.user/442D8A65/sources/s-F115D376/lock_file]; OCCURRED AT: virtual rstudio::core::Error rstudio::core::AdvisoryFileLock::acquire(const rstudio::core::FilePath &) /Users/vagrant/workspace/IDE/macos-v1.2/src/cpp/core/file_lock/AdvisoryFileLock.cpp:139; LOGGED FROM: bool rstudio::session::source_database::supervisor::(anonymous namespace)::reclaimOrphanedSession() /Users/vagrant/workspace/IDE/macos-v1.2/src/cpp/session/SessionSourceDatabaseSupervisor.cpp:374
24 Jun 2019 16:24:51 [rsession-<user>] ERROR system error 45 (Operation not supported) [lock-file=<remote-path>/.Rproj.user/442D8A65/sources/s-F115D376/lock_file]; OCCURRED AT: virtual rstudio::core::Error rstudio::core::AdvisoryFileLock::acquire(const rstudio::core::FilePath &) /Users/vagrant/workspace/IDE/macos-v1.2/src/cpp/core/file_lock/AdvisoryFileLock.cpp:139; LOGGED FROM: rstudio::core::Error rstudio::session::source_database::supervisor::(anonymous namespace)::createSessionDir() /Users/vagrant/workspace/IDE/macos-v1.2/src/cpp/session/SessionSourceDatabaseSupervisor.cpp:256
```

The timestamps look like they're from a completely different time zone, but the minutes match up to when i load the file.

**Comment by ronblum** [comment]
@baderstine Thank you for raising this. We've seen other issues with remote servers, although I can't find one that's quite like this. I'll mark it as a bug to review. Did this issue occur with earlier releases, such as 1.1.463, or just 1.2.1335?

**Comment by baderstine** [comment]
I only noticed it after updating from 1.1.456 to 1.2.1335

**Comment by eliyara** [comment]
I have the same issues, tried older versions, they fail differently, showing an error message. I used the same version couple of weeks ago, and today it just showed the above message!

**Comment by kevinushey** [comment]
I believe this is because advisory file locking isn't supported by the networked filesystem you're using. You can try placing the following in a file at `/etc/rstudio/file-locks`:

```
lock-type=linkbased
```

Then, restart RStudio and see if things work better.

**Comment by kevinushey** [comment]
Sorry -- that won't help. We always create advisory file locks in that directory when using RStudio Desktop :-/

**Comment by Jero327** [comment]
Try to reload R, specific it's version 32 or 64.

**Comment by baderstine** [comment]
Possibly related: 
After loading the remote .Rproj I cannot perform a search the Rstudio Help pane. I type a term into the search box and after pressing Enter, get a "R code execution error" popup and an error message in the console:

```r 
Error in readRDS(f) : unknown input format
> traceback()
4: readRDS(f)
3: index.search(topic, find.package(if (is.null(package)) loadedNamespaces() else package, 
       lib.loc, verbose = verbose))
2: help(query, help_type = "html")
1: (function (query) 
   {
       exactMatch = help(query, help_type = "html")
       if (length(exactMatch) == 1) {
           print(exactMatch)
           return()
       }
       else {
           paste("help/doc/html/Search?pattern=", utils::URLencode(query, 
               reserved = TRUE), "&title=1&keyword=1&alias=1", sep = "")
       }
   })("names")
``` 

I am still trying to figure out what exact steps I need to take to reproduce the error. Usually, I can correct the issue by restarting R (`Session > Restart R`).   I am currently finding that the behavior is inconsistent and possibly related to which packages I have loaded and in what order I loaded them.

**Comment by kevinushey** [comment]
Odds are one of your installed packages is corrupt. R is attempting to read a help database file (normally in `.rds` format) but that is failing.

**Comment by baderstine** [comment]
Seems to be `dplyr`.  Following steps will reproduce it:

```
library(dplyr)
help("test")

mtcars
help("test")

mtcars_tibble <- mtcars %>% 
  group_by(cyl) %>% 
  select(cyl, hp, mpg) %>% 
  distinct()
help("test")

mtcars_tibble
help("test")
```

Only after printing the tibble to the console, I cannot perform help searches:

```
> help("test")
Error in readRDS(f) : unknown input format
```

**Comment by kevinushey** [comment]
You could also use e.g.

```
options(error = recover)
```

and then using the R debugger to introspect which file R was attempting to read.

**Comment by baderstine** [comment]
Aha, that would be the following:

```
1: help("test")
2: index.search(topic, find.package(if (is.null(package)) loadedNamespaces() else package, lib.loc, verbose = verbos
3: readRDS(f)

Selection: 3
Called from: index.search(topic, find.package(if (is.null(package)) loadedNamespaces() else package, 
    lib.loc, verbose = verbose))
Browse[1]> file
[1] "/Library/Frameworks/R.framework/Versions/3.5/Resources/library/fansi/help/aliases.rds"
```

**Comment by baderstine** [comment]
These errors might also be related to extremely slow performance that is observed when working from network drive files. Oddly the behavior is unpredictable and sometimes is very slow returning trivial requests (e.g., 1+1).  I have seen this behavior reported [elsewhere](https://community.rstudio.com/t/rstudio-desktop-slow-and-laggy-when-project-is-on-a-network-drive/7416).  
Using the suggested `` Ctrl+` ``combination...  the longest events seen are: `get_events`,  and `save_document_diff`.  Auto-complete is so slow and delayed most of the time that I often complete typing the thing that I want to complete before the tab-complete returns, and then end up with gibberish. The related event is a rather long `get_completions`.

**Comment by baderstine** [comment]
@kevinushey is there some way I can test (e.g., via command line) the networked file system to see what it does and does not support re: advisory file locking?

**Comment by kevinushey** [comment]
You could try testing with the `filelock` package. Here's some example code:

```

library(filelock)

# acquire lock in this process
file <- tempfile("lockfile-")
lock <- filelock::lock(file)
print(lock)

# try acquiring lock on same file in sub-process
code <- substitute({
  library(filelock)
  filelock::lock(file, timeout = 0)
}, list(file = file))

# write to separate script and run
script <- tempfile("lock-", fileext = ".R")
writeLines(deparse(code), con = script)
R <- file.path(R.home("bin"), "R")
system2(R, c("-s", "-f", shQuote(script)))
```

I see:

```
> source('~/scratch/lock.R')
Lock on '/private/tmp/RtmpEF41hr/lockfile-101954119ff1b'
NULL
```

The output of `NULL` is expected if the process was unable to acquire the lock.

**Comment by baderstine** [comment]
I get similar responses both in a local folder: 
```
> source('~/scratch/test_filelock.R')
Lock on ‘/private/var/folders/5m/gd6x5rh95d993ty7gx1lw9y80000gq/T/RtmpMKVxVx/lockfile-332971aa1bae’
NULL
```
and when running from a network drive folder: 
```
> source('/Volumes/<snipped>/test_filelock.R')
Lock on ‘/private/var/folders/5m/gd6x5rh95d993ty7gx1lw9y80000gq/T/RtmpMKVxVx/lockfile-33291dd0fb01’
NULL
```

**Comment by kevinushey** [comment]
Thanks for testing! I think that rules out file locking being the issue, since it looks like everything is working as expected...

Have you already tried testing with the latest RStudio release (1.4)?

**Comment by baderstine** [comment]
Tested with latest release (1.4.1103) the past few days.  The error messages do not appear and the project loads.  It takes a a long time, but it works so I'm satisfied.

**Comment by gadepallivs** [comment]
We have this network issue on the windows. Its very slow, the path does not open to the folder the .Rproject file is saved in.  When I open the .Rproj file after sometime on the "initializing screen"  I get this error
![image](https://user-images.githubusercontent.com/13947738/117013382-d0fd7500-acbd-11eb-8ab5-507a4ab15e80.png)

When I click ok, I get a notification "project closing" and the Rstudio still remains open, but project folder opens to a path that is different. 

To manually change the working directory, I run this code ( source Stack overflow)  to change the directory. 

```
change_here <- function(new_path){
  new_root <- here:::.root_env
  new_root$f <- function(...){file.path(new_path, ...)}
  assignInNamespace(".root_env", new_root, ns = "here")
}
change_here("The desired directory on the network") 
```

This process of manually changing to the desired directory on network used to work earlier but recently I updated the Rstudio  to 1.4 and it is not working any more. 

```
Session info --------------------------------------------------
 setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United States.1252  
 ctype    English_United States.1252  
 tz       America/New_York            
 date     2021-05-04  

```

**Comment by ronblum** [comment]
@baderstine Thank you for confirming this!

@gadepallivs Thank you for raising this—I moved it to a new ticket, #9304.
